lib/melodica_inventory/goods/image_operations.ex

```diff
@@ -0,0 +1,50 @@
+defmodule MelodicaInventory.Goods.ImageOperations do
+  @moduledoc false
+
+  import Ecto.Query
+
+  alias MelodicaInventory.Goods.{Item, Image}
+  alias MelodicaInventory.Repo
+
+  def upload_images(item_id, nil) do
+    {:ok, []}
+  end
+
+  def upload_images(item_id, uploaded_files) do
+    uploaded_files
+    |> Enum.each(&upload_to_cloudex_and_save(&1, item_id))
+
+    {:ok, []}
+  end
+
+  def delete_images_from_cloudex([]), do: {:ok, true}
+
+  def delete_images_from_cloudex(images) do
+    images
+    |> Enum.map(&(&1.public_id))
+    |> Cloudex.delete()
+    |> Enum.reduce(fn ({:ok, _}), _ -> {:ok, true} end)
+  end
+
+  def delete_images(images) do
```

Comment:
Let's rename this argument to `public_ids` or `image_public_ids` as it is not obvious right not that we are handling a list of ids here.